packages/flutter_mobx_hooks/test/use_observer_test.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_mobx_hooks/flutter_mobx_hooks.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'fake_store.dart';
+
+void main() {
+  testWidgets('useObserver', (tester) async {
+    var value = 0;
+    final store = Counter();
+
+    await tester.pumpWidget(HookBuilder(
```

Comment:
This needs to test more advanced scenarios, such as two nested HookBuilder, both using useObserver but observing different thing, and checking that rebuilding one doesn't rebuild the other